### PR TITLE
[MAINTENANCE] Remove `notify_on` from base Action

### PR DESCRIPTION
**Pull request by cdkini**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/checkpoint/actions.py [hunk 1]
```diff
@@ -147,6 +137,16 @@ def _get_data_docs_pages_from_prior_action(
...
 
 
+def _is_enabled(success: bool, notify_on: Literal["all", "failure", "success"]) -> bool:
+    return (
+        notify_on == "all"
+        or notify_on == "success"
+        and success
+        or notify_on == "failure"
+        and not success
+    )
```
Comment (cdkini) [inline great_expectations/checkpoint/actions.py]: I don't love it but a shared utility is the easiest way to get this functional - could subclass for conditional actions but that feels drastic for what should be a very small change

File: great_expectations/checkpoint/actions.py [hunk 2]
```diff
@@ -147,6 +137,16 @@ def _get_data_docs_pages_from_prior_action(
         return None
 
 
+def _is_enabled(success: bool, notify_on: Literal["all", "failure", "success"]) -> bool:
```
Comment (tyler-hoffman) [inline great_expectations/checkpoint/actions.py]: Small but blocking: this looks like it should probably be renamed to something like `_should_notify`.

File: great_expectations/checkpoint/actions.py [hunk 3]
```diff
@@ -782,6 +782,7 @@ class SNSNotificationAction(ValidationAction):
 
     type: Literal["sns"] = "sns"
 
+    notify_on: Literal["all", "failure", "success"] = "all"
```
Comment (tyler-hoffman) [inline great_expectations/checkpoint/actions.py]: Do this class and the next not actually make use of this field? I don't see it used, and I don't see it on them in 0.18.

**Review by tyler-hoffman — CHANGES_REQUESTED** [review]

**Review by tyler-hoffman — APPROVED** [review]
Sweet. Thank you!

**Pull request merged**